packages/backend-function/src/factory.ts

```diff
@@ -150,6 +151,12 @@ export type FunctionProps = {
    * Options for bundling the function code.
    */
   bundling?: FunctionBundlingOptions;
+
+  /**
+   * Group the function with existing Amplify resources or separate the function into its own group.
+   * @default 'function' // grouping with other Amplify functions
+   */
```

Comment:
@josefaidt , my 2 cents, start with "Optional ... " and include recommendations that cover base use cases, ie "if you want to add an auth trigger... set this to... " etc

Reply:
Good point, added an example.

Reply:
looking at the default, is "function" a valid value?

Reply:
"function" is what we currently have as the resourceGroupName in our construct factory, see https://github.com/aws-amplify/amplify-backend/blob/main/packages/backend-function/src/factory.ts#L342. This change will update the value if customers pass in a value for `resourceGroupName`, otherwise it falls back to default behavior.